paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -107,9 +133,103 @@ def paasta_mark_for_deployment(args):
     if service and service.startswith('services-'):
         service = service.split('services-', 1)[1]
     validate_service_name(service, soa_dir=args.soa_dir)
-    return mark_for_deployment(
+    ret = mark_for_deployment(
         git_url=args.git_url,
         deploy_group=deploy_group,
         service=service,
         commit=args.commit,
     )
+    if args.block:
+        try:
+            _log(
+                service=service,
+                line="Waiting for deployment of {0} to {1} complete".format(args.commit, deploy_group),
+                component='deploy',
+                level='event'
+            )
+            wait_for_deployment(args.service, args.deploy_group,
+                                args.commit, args.soa_dir, args.timeout)
+            _log(
+                service=service,
+                line="Deployment of {0} to {1} complete".format(args.commit, deploy_group),
+                component='deploy',
+                level='event'
+            )
+        except TimeoutError:
+            sys.exit(1)
+    return ret
+
+
+def are_instances_deployed(cluster, service, instances, git_sha):
+    api = client.get_paasta_api_client(cluster=cluster)
+    if not api:
+        # Assume not deployed if we can't reach API
+        return False
+    statuses = []
+    for instance in instances:
+        try:
+            statuses.append(api.service.status_instance(service=service, instance=instance).result())
+        except HTTPError as e:
+            if e.response.status_code == 404:
+                line = "Can't get status for instance {0}, service {1} in cluster {2}. "\
+                       "This is normally because it is a new service that hasn't been "\
+                       "deployed by PaaSTA yet".format(instance, service, cluster)
+                _log(
```

Comment:
This is going to be pretty noisy. We are talking quite a few lines every 10 seconds. I think it would be ok to just "print" this stuff